src/fsharp/TypedTreeOps.fs

```diff
@@ -2243,7 +2269,10 @@ and accFreeInTypeLeftToRight g cxFlag thruFlag acc ty =
     | TType_app (_, tinst) -> 
         accFreeInTypesLeftToRight g cxFlag thruFlag acc tinst 
     | TType_ucase (_, tinst) -> 
-        accFreeInTypesLeftToRight g cxFlag thruFlag acc tinst 
+        accFreeInTypesLeftToRight g cxFlag thruFlag acc tinst
+    // SWOORUP TODO: No idea wtf this is
+    | TType_erased_union (_, tinst) -> 
+        accFreeInTypesLeftToRight g cxFlag thruFlag acc tinst
```

Comment:
What you have is correct - it's just finding all the free (unsolved) type variables throught the type in canonical left-to-right order